### ISSUE TITLE
Not able to use gRPC without TLS using .net core

### ISSUE THREAD

**Issue by gabrielamartinezatwizeline** [grpc/grpc-dotnet#1042]
On Windows 10, I am using the template project for gRPC in ASP.NET Core  3.1.
I have a client and a server app that run perfectly fine using TLS. The issue appears when I try to use http instead.  I have tried the recommendations on the link bellow for the troubleshooting.

https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-3.0#call-insecure-grpc-services-with-net-core-client

Also I looked at https://stackoverflow.com/questions/58682933/how-do-i-enable-http2-in-c-sharp-kestrel-web-server-over-plain-http and did all that they mentioned.

The issue is happening in the Client when communicating with the server. Here's an extract of the code from the client

```
            GrpcChannelOptions options = new GrpcChannelOptions();
            string baseUrl = "http://localhost:8000/";
            HttpClient httpClient = new HttpClient() { BaseAddress = new Uri(baseUrl), DefaultRequestVersion = new Version(2,0)};
            
            options.HttpClient = httpClient;
            options.LoggerFactory = _loggerFactory;
            options.Credentials = ChannelCredentials.Insecure;

            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
            var channel = GrpcChannel.ForAddress(baseUrl, options);

            return new Data.DataClient(channel);
```

Here's the log of the error 

```
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.7\Microsoft.Win32.Registry.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Grpc.Net.Client.Internal.GrpcCall: Debug: gRPC call canceled.
Exception thrown: 'Grpc.Core.RpcException' in System.Private.CoreLib.dll
Exception thrown: 'Grpc.Core.RpcException' in System.Private.CoreLib.dll
Grpc.Net.Client.Internal.GrpcCall: Error: Call failed with gRPC error status. Status code: 'Internal', Message: 'Error starting gRPC call: An error occurred while sending the request.'.
Exception thrown: 'Grpc.Core.RpcException' in System.Private.CoreLib.dll
Grpc.Net.Client.Internal.GrpcCall: Debug: Finished gRPC call.
Exception thrown: 'Grpc.Core.RpcException' in System.Private.CoreLib.dll
Exception thrown: 'Grpc.Core.RpcException' in System.Private.CoreLib.dll
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.7\System.IO.MemoryMappedFiles.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Microsoft.AspNetCore.Server.IIS.Core.IISHttpServer: Error: Connection ID "18230571297501348041", Request ID "800000ca-0001-fd00-b63f-84710c7967bb": An unhandled exception was thrown by the application.

Grpc.Core.RpcException: Status(StatusCode=Internal, Detail="Error starting gRPC call: An error occurred while sending the request.")
   at Grpc.Net.Client.Internal.HttpContentClientStreamReader`2.MoveNextCore(CancellationToken cancellationToken)
   at Grpc.Core.AsyncStreamReaderExtensions.ReadAllAsync[T](IAsyncStreamReader`1 streamReader, CancellationToken cancellationToken)+MoveNext()
   at Grpc.Core.AsyncStreamReaderExtensions.ReadAllAsync[T](IAsyncStreamReader`1 streamReader, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at GrpcClient.Controllers.DataController.GetAsyncEnumerable()+MoveNext() in C:\Wizelinemigration\repository\wizeline-factset-datahub-api\DatahubAPI\GrpcClient\Controllers\DataController.cs:line 47
   at GrpcClient.Controllers.DataController.GetAsyncEnumerable()+MoveNext() in C:\Wizelinemigration\repository\wizeline-factset-datahub-api\DatahubAPI\GrpcClient\Controllers\DataController.cs:line 47
   at GrpcClient.Controllers.DataController.GetAsyncEnumerable()+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at GrpcClient.Controllers.DataController.Get()+MoveNext() in C:\Wizelinemigration\repository\wizeline-factset-datahub-api\DatahubAPI\GrpcClient\Controllers\DataController.cs:line 63
   at GrpcClient.Controllers.DataController.Get()+MoveNext() in C:\Wizelinemigration\repository\wizeline-factset-datahub-api\DatahubAPI\GrpcClient\Controllers\DataController.cs:line 63
   at GrpcClient.Controllers.DataController.Get()+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at Microsoft.AspNetCore.Mvc.Infrastructure.AsyncEnumerableReader.ReadInternal[T](Object value)
   at Microsoft.AspNetCore.Mvc.Infrastructure.AsyncEnumerableReader.ReadInternal[T](Object value)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor.ExecuteAsyncEnumerable(ActionContext context, ObjectResult result, Object asyncEnumerable, Func`2 reader)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeResultFilters>g__Awaited|27_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContextOfT`1.ProcessRequestAsync()
The thread 0x19c8 has exited with code 0 (0x0).
```

The same solutions work fine in TLS, but I also want to be able to use it without a certificate. 

Thanks in advance for your time!

**Comment by JamesNK** [comment]
You need to move `AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);` so it is called before the `HttpClient` is created.

**Comment by gabrielamartinezatwizeline** [comment]
thanks @JamesNK  for the suggestion, I did that but the app was not working until I added this part:

```
 webBuilder.ConfigureKestrel(options =>
            {
               // Setup a HTTP/2 endpoint without TLS.
                options.ListenLocalhost(5000, o => o.Protocols = 
                    HttpProtocols.Http2);
            });
```

From the docs:

https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-3.1#unable-to-start-aspnet-core-grpc-app-on-macos

My only suggestion is it would be great to make this clearer in the docs. Since this is inside a section that says `on macOs`  meaning that this only applies to mac. But I am working on Windows 10 and still had to add it to the app in order to make it work with Http. 
Thanks a lot for your taking a look on this!
Closing this issue,since there is no needed any other assistance or bug.
